### PR TITLE
Implemented JWT support and added login endpoint

### PR DESCRIPTION
**Pull request by dima-semeniuk**

### CODE REVIEW

File: src/main/java/mate/academy/webapp/security/JwtUtil.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public String generateToken(String username) {
+        return Jwts.builder()
+                .setSubject(username)
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + expiration))
+                .signWith(secret)
+                .compact();
+    }
+
+    public boolean isValidToken(String token) {
```
Comment (bhdnchui) [inline src/main/java/mate/academy/webapp/security/JwtUtil.java:35]: Please, rename. For example `isTokenValid`

File: src/main/java/mate/academy/webapp/dto/user/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package mate.academy.webapp.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+
```
Comment (bhdnchui) [inline src/main/java/mate/academy/webapp/dto/user/UserLoginRequestDto.java:8]:
```suggestion
```

File: src/main/java/mate/academy/webapp/config/SecurityConfig.java [hunk 3]
```diff
@@ -1,23 +1,33 @@
 package mate.academy.webapp.config;
```
Comment (bhdnchui) [inline src/main/java/mate/academy/webapp/config/SecurityConfig.java:1]: It's better to place this file in a `security` package.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
A few minor points to consider.

File: src/main/java/mate/academy/webapp/controller/BookController.java
Comment (d1sam) [inline src/main/java/mate/academy/webapp/controller/BookController.java:1]: why did you remove `hasRole('ADMIN')`?

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**